library/bignum.c

```diff
@@ -1612,21 +1545,9 @@ int mbedtls_mpi_mod_int( mbedtls_mpi_uint *r, const mbedtls_mpi *A, mbedtls_mpi_
     return( 0 );
 }
 
-/*
- * Fast Montgomery initialization (thanks to Tom St Denis)
- */
 static void mpi_montg_init( mbedtls_mpi_uint *mm, const mbedtls_mpi *N )
```

Comment:
Why not go one step further and elimitate this trivial auxiliary function, as was done for `mpi_sub_hlp`? Likewise for `mpi_montmul`.

Reply:
To my understanding RR and mm will be pre-calcuated during [mbedtls_mod_modulus_setup()](https://github.com/Mbed-TLS/mbedtls/blob/d932de8857a5e9861448858776cde1ae419e57f5/library/bignum_mod.c#L104-L140) phase. 

There is also the consideration of adjusting them to a more efficient computionally logic. So it makes it simpler to keep them as small single purpose service functions, which can be absorbed later on when the new interface is the standard.

Reply:
> Why not go one step further and eliminate this trivial auxiliary function

Very specifically, for this PR to have minimal, obvious, easily-reviewable changes. The legacy bignum interface will be retired at some point, and then these will just vanish.

Reply:
I don't think it is so trivial, mathematically it is quite involved. I would prefer to have it in a separate function and document it better in the future.

Reply:
I didn't mean to eliminate `mbedtls_mpi_montg_init`, which is where the mathematics happen. I meant to eliminate _this_ function, `mpi_montg_init`, and call `mbedtls_mpi_montg_init` directly.

Reply:
As I said above, my goal was for this PR to have minimal, obvious, easily-reviewable changes. Making the existing function marshal its arguments and call the new function is (to me) the minimal, obvious, way to do this.

Since `mpi_montg_init()` is static, the compiler inlines it (at least on `-O2`), and there is no code-size impact.

And in the future, all of this is going away.

So I don't see the benefit of changing the caller of `mbedtls_mpi_montg_init()`